src/index.js

```diff
@@ -6,46 +6,55 @@ const tags = tagsW3.map(tag => tag.replace(/[<|>]/g, ""));
 
 // Wait for the DOM to be fully loaded
 document.addEventListener("DOMContentLoaded", function () {
-    uploadTagsFromLocalStorage(); // list all tags in localStorage
-    updateProgressBar();
-    // Find the form element and add a submit event listener
-    const form = document.querySelector("form");
-    form.addEventListener("submit", function (event) {
-        // on submit
-        event.preventDefault(); // Prevent the default form submission
-        submitVal(); // call our submitVal function
-    });
-
-    function submitVal() {
-        let val = document.getElementById("input1").value;
-        // remove all white spaces and non-alphabetical characters except . ! - 
-        let regex = /[^a-zA-Z!-.]/g;
-        // change input to lower case and apply regex
-        let sanitizedVal = val.toLowerCase().replace(regex, "");
-
-        let li = "<li>" + sanitizedVal + "</li>";
-        // check if the input value is in the tags array
-        let found = tags.some(tag => tag.toLowerCase() === sanitizedVal);
-        if (found) {
-            document.getElementById("list").insertAdjacentHTML("beforeend", li);
-            saveTagToLocalStorage(sanitizedVal); // Save tag to localStorage
-            updateProgressBar();
-        }
-        document.getElementById("input1").value = "";
+
+  uploadTagsFromLocalStorage(); // list all tags in localStorage
+  updateProgressBar();
+  // Find the form element and add a submit event listener
+  const form = document.querySelector("form");
+  form.addEventListener("submit", function (event) {
+    // on submit
+    event.preventDefault(); // Prevent the default form submission
+    submitVal(); // call our submitVal function
+  });
+
+  function submitVal() {
+    let val = document.getElementById("input1").value;
+    // remove all white spaces and non-alphabetical characters except . ! - 
+    let regex = /[^a-zA-Z!-.]/g;
+    // change input to lower case and apply regex
+    let sanitizedVal = val.toLowerCase().replace(regex, "");
+    let li = "<li>" + sanitizedVal + "</li>";
+    // check if the input value is in the tags array
+    let found = tags.some(tag => tag.toLowerCase() === sanitizedVal);
+    // get the list of inputs already made in JSON format
+    const getJSON = JSON.stringify(localStorage.getItem("myTags"));
+    // convert the list to an array
+    const getList = JSON.parse(getJSON);
+
```

Comment:
It works! Nice!

One suggestion though to simplify your code: you don't need line 30. The tags are pulled from local storage so they are already in string form (because before they are saved we already use JSON.stringify on them). So all you need is line 32:
JSON.parse(localStorage.get ...)